accepted/0000-workspace-run-commands.md

```diff
@@ -0,0 +1,57 @@
+- Start Date: 2017-09-21
+- RFC PR:
+- Yarn Issue:
+
+# Summary
+
+Allow Yarn CLI to execute a package script on each packages from the workspace root.  
+
+# Motivation  
+
+"Lerna". Lerna does a great job for handling monorepos. Since Yarn has a built-in 
+workspaces feature, we could use some of the functionalities that being used in 
+lerna (not entirely though!) for handling the monorepo more efficiently.  
+
+Just like installing all dependencies from one place (workspace root), it'd be great 
+to execute all the sub-package scripts from the workspace root using a single command. 
+Not only this would make a less back-and-forth traveling between the packages to 
+execute scripts, but also it'll help a lot for CI/CD configuration. 
+
+
+# Detailed design  
+
+As @BYK mentioned [here](https://github.com/yarnpkg/yarn/issues/4467#issuecomment-330873337), 
+we could create two commands for this feature.  
+
+* `yarn workspaces list`  
+* `yarn workspaces run <command>`
+
+By introducing `workpaces` command, it brings a modular approach from the CLI's perspective.  
+
+## `yarn workspaces list`  
+
+This command lists out the package names inside that workspace.  In the case of multiple workspace configuration, we'll need to pass the workspace name like `yarn workspaces <workspace_name> list`. By defaut, when no workspace name is not given, it'll pick the first workspace name from the root package.json.  
```

Comment:
Can you clarify what you mean by a `multi-workspace` config?

Reply:
I'm so sorry.. Got confused with this. 
![image](https://user-images.githubusercontent.com/2258588/30705412-04487868-9f13-11e7-8159-076a9a915f97.png) I thought we had a multi-workspace setting. Will remove that.

Reply:
Seems more logical to either list all packages for all workspaces or fail with error.
Picking first workspace is strange and not really useful

Reply:
+1 to listing all of the packages in all the workspaces. That would also allow for easy "filtering" of workspaces if you only want to list a specific few of them, like:

```
yarn workspaces --filter workspace-a list
```

...or similar. (I think something like `--filter` functionality would ideally be available on all the `workspaces` commands.)

Reply:
@ianstormtaylor thanks for that `--filter` tip.